pkg/ucp/store/client.go

```diff
@@ -77,8 +88,55 @@ type Query struct {
 	Filters []QueryFilter
 }
 
+// Validate validates the Query.
+func (q Query) Validate() error {
+	var err error
+	if q.RootScope == "" {
+		err = errors.Join(err, &ErrInvalid{Message: "RootScope is required"})
```

Comment:
For reviewers - `errors.Join` is relatively new in go and useful for places like this. Instead of reporting the first error, we can report all of them.